tests/lit-tests/ssp-asm-linux.ispc

```diff
@@ -0,0 +1,23 @@
+// Make sure the backend honors the request to generate code to detect stack smashing.
+// We check for a call to __stack_chk_fail.
+// RUN: %{ispc} -O0 --emit-asm %s --target-os=linux --stack-protector=strong -o - | FileCheck %s
+// RUN: %{ispc} -O1 --emit-asm %s --target-os=linux --stack-protector=strong -o - | FileCheck %s
+// RUN: %{ispc} -O2 --emit-asm %s --target-os=linux --stack-protector=strong -o - | FileCheck %s
+// RUN: %{ispc} -O3 --emit-asm %s --target-os=linux --stack-protector=strong -o - | FileCheck %s
+// REQUIRES: LINUX_ENABLED
```

Comment:
I suggest having just one test for "host" os that universally works on ARM/x86 and Linux/Windows/macOS. Regexp syntax for matching expression is flexible enough to allow matching both `call` and `bl` instruction, as well as all variations of security functions that we aware of. If we see fail on other platforms, we'll just add more variations later.

If you specifically care about cross-compilation scenario (I suspect you do), it's ok to have one more test for cross compilation scenario (i.e. `--target-os=<your os>`) that you care about. But it's up to you. I think if `host` is works, we should be good.